configure.ac

```diff
@@ -5239,9 +5239,9 @@ WITH_SAVE_ENV([
         #include <unistd.h>
       ], [
         #ifdef HAVE_CRYPT_R
-          void *x = crypt_r;
+          volatile void *x = crypt_r;
```

Comment:
Unfortunately this is not enough.  Making an auto variable volatile doesn't prevent the optimizer from removing it.  I tried this change first so I know it doesn't work :).  That's why in my suggested change, I made `x` a global volatile variable instead.

Reply:
Oh. I modified my PR to avoid volatile, but use the variable value instead.

Please check the updated PR.